### ISSUE TITLE
empty file list when asset directory is in nested folder

### ISSUE THREAD

**Issue by yatesco** [pyros2097/rust-embed#99]
Hi,

Given:
```
\ (the root of cargo workspace)
\Cargo.toml
\assets\ (a simple folder)
\component-a\ (a crate within the workspace)
\component-a\Cargo.toml
\component-a\assets
\static\ (a simple folder)
\static\assets
```

where each "\assets" directory is a copy with at least `a.txt` in, then `#[folder = "assets/"]` is the only one that works when used within `\component-a`.

Observations:
 - `#[folder = "component-a/assets"]` compiles but lists no files
 - `#[folder = "static/assets"]` compiles but lists no files

this is regardless of whether `cargo run` is run within `\` or `\component-a`.

What I am trying to achieve is either `component-a/assets/` or `static/assets/`.

Help :-)

**Comment by yatesco** [comment]
Actually, it is slightly more subtle: `\assets\` must exist but it can be empty, the _actual_ assets are loaded from `\component-a\assets`.

If `\assets\` is missing then it won't compile. If `\component-a\assets\` is missing then it will compile but not find anything.

**Comment by yatesco** [comment]
Nope...event more subtle ;-). For developing `\assets\` must exist but the actual assets come from `\component-a\assets` but for a `release` build `\assets\` is used :-(.

**Comment by yatesco** [comment]
Last comment (for now :-)).

Synchronicity (and thus sanity!) can be achieved by enabling the feature "debug-embed" which removes the need for `\component-a\assets` but does have the cost of needing a rebuild for each change to an asset.

**Comment by Mcat12** [comment]
Can you provide a link to your code? Your explanation is hard to follow. Are you using a symlink?

**Comment by yatesco** [comment]
No symlinks (this is on Windows) and I’ll get a reproducible repo online tomorrow:-)

**Comment by yatesco** [comment]
Actually, the whole thing is simpler than I thought, and is related to #63 

_Regardless_ of whether using `debug` or `release`, the value in the `folder` directive is relative to the current directory, which is really confusing ;-). It would be nice if it could _always_ be relative to the workspace root.

Repo is at https://github.com/yatesco/rust-embed-bug-with-workspaces

**Comment by yatesco** [comment]
Note to self: check if https://github.com/pyros2097/rust-embed/pull/101/commits/bbe6eea0a8dd415942b6a8cef977d0b2e4f0fd13 fixes this.

**Comment by Mcat12** [comment]
@yatesco Is this fixed for you as of #101?

**Comment by Mcat12** [comment]
Tested the [reproduction code](https://github.com/yatesco/rust-embed-bug-with-workspaces) with the latest version, and the issue is fixed. The embedded path no longer depends on where you start `cargo run`.
